forestserviceprototype/specialuseform/views.py

```diff
@@ -1,21 +1,56 @@
 from django.contrib.auth.decorators import login_required
 from django.shortcuts import render, redirect, get_object_or_404
 from django.http import HttpResponse
+from django.forms.models import model_to_dict
 from .forms import PermitForm
 from .models import Permit
+import logging
 
 # Create your views here.
+logger = logging.getLogger(__name__)
 
-def submit(request):
-    form = PermitForm(request.POST or None)
-    if form.is_valid():
-        permit = form.save()
-        # send_report_notification(request, permit)
-        return redirect(submitted_permit)
-    return render(request, "specialuseform/submit.html", {'form': form})
 
-def submitted_permit(request):
-    return render(request, "specialuseform/submitted_permit.html")
+# def submit(request):
+#     form = PermitForm(request.POST or None)
+#     if form.is_valid():
+#         permit = form.save()
+#         # send_report_notification(request, permit)
+#         return redirect(submitted_permit, permit_id = permit.permit_id)
+#     return render(request, "specialuseform/submit.html", {'form': form})
+
+def submit(request, permit_id=None, template_name='specialuseform/submit.html'):
+    if permit_id:
+        permit = get_object_or_404(Permit, permit_id=permit_id)
+        submit_button_text = 'Save Changes'
+    else:
+        permit = Permit()
+        submit_button_text = 'Submit Your Application'
+
+    form = PermitForm(request.POST or None, instance=permit)
+    if request.POST and form.is_valid():
+        form.save()
+
+        # Save was successful, so redirect to another page
+        # return redirect(submitted_permit, permit.permit_id)
```

Comment:
I haven't yet tested the functionality, but assuming it's working, line 34 can be removed.